monkey/infection_monkey/monkey.py

```diff
@@ -401,35 +402,63 @@ def _send_log():
+            return False
+
+        try:
+            if "win32" == sys.platform:
+                InfectionMonkey._self_delete_windows()
+            else:
+                InfectionMonkey._self_delete_linux()
+
+            T1107Telem(ScanStatus.USED, sys.executable).send()
+            return True
+        except Exception as exc:
+            logger.error("Exception in self delete: %s", exc)
+            T1107Telem(ScanStatus.SCANNED, sys.executable).send()
+
+        return False
+
+    @staticmethod
+    def _remove_monkey_dir():
         status = ScanStatus.USED if remove_monkey_dir() else ScanStatus.SCANNED
         T1107Telem(status, get_monkey_dir_path()).send()
-        deleted = False
-
-        if -1 == sys.executable.find("python"):
-            try:
-                status = None
-                if "win32" == sys.platform:
-                    from subprocess import CREATE_NEW_CONSOLE, STARTF_USESHOWWINDOW, SW_HIDE
-
-                    startupinfo = subprocess.STARTUPINFO()
-                    startupinfo.dwFlags = CREATE_NEW_CONSOLE | STARTF_USESHOWWINDOW
-                    startupinfo.wShowWindow = SW_HIDE
-                    subprocess.Popen(
-                        DELAY_DELETE_CMD % {"file_path": sys.executable},
-                        stdin=None,
-                        stdout=None,
-                        stderr=None,
-                        close_fds=True,
-                        startupinfo=startupinfo,
-                    )
-                    deleted = True
-                else:
-                    os.remove(sys.executable)
-                    status = ScanStatus.USED
-                    deleted = True
-            except Exception as exc:
-                logger.error("Exception in self delete: %s", exc)
-                status = ScanStatus.SCANNED
-            if status:
-                T1107Telem(status, sys.executable).send()
-            return deleted
+
+    @staticmethod
+    def _self_delete_windows():
+        from subprocess import CREATE_NEW_CONSOLE, STARTF_USESHOWWINDOW, SW_HIDE
+
+        # Time for delay deleting monkey executable
+        DELAY_SECONDS = 5
+
+        monkey_pid = os.getpid()
+        monkey_file_path = sys.executable
+        # Command that returns 1 if the process is running and 0 otherwise
+        CHECK_RUNNING_MONKEY_CMD = f'tasklist /fi "PID eq {monkey_pid}" ^| find /C "{monkey_pid}"'
+        DELETE_FILE_AND_EXIT = f"del /f /q {monkey_file_path} & exit"
+        # Command that checks for running monkey process 20 times
+        # If the monkey is running it sleeps for 'delay_seconds'
+        # If the monkey is not running it deletes the executable and exits the loop
+        DELAY_DELETE_CMD = (
+            f'cmd /c (for /l %i in (1,1,20) do (for /F "delims=" %j IN '
+            f'(\'{CHECK_RUNNING_MONKEY_CMD}\') DO if "%j"=="1" (timeout {DELAY_SECONDS}) else '
+            f"({DELETE_FILE_AND_EXIT})) ) > NUL 2>&1"
+        )
```

Comment:
These don't need to be all caps anymore

Reply:
Maybe wrap this all in a function called 'build_windows_delete_command()` or something similar. Have it return a string.